mate-dictionary/src/gdict-applet.c

```diff
@@ -352,6 +360,14 @@ gdict_applet_build_window (GdictApplet *applet)
   bbox = gtk_button_box_new (GTK_ORIENTATION_HORIZONTAL);
   gtk_button_box_set_layout (GTK_BUTTON_BOX (bbox), GTK_BUTTONBOX_END);
   gtk_box_set_spacing (GTK_BOX (bbox), 6);
+
+#if defined (ENABLE_WAYLAND) && defined (GDK_WINDOWING_WAYLAND)
+  if (GDK_IS_WAYLAND_DISPLAY (display))
+  {
+    gtk_box_pack_start (GTK_BOX (bbox), priv->entry, TRUE, TRUE, 0);
+    gtk_button_box_set_child_non_homogeneous(GTK_BOX (bbox), priv->entry, TRUE);
```

Comment:
@lukefromdc 
I found some serious build  warnings if build in-process.
```
gdict-applet.c: In function 'gdict_applet_build_window':
/usr/include/glib-2.0/gobject/gtype.h:2536:6: warning: passing argument 1 of 'gtk_button_box_set_child_non_homogeneous' from incompatible pointer type [-Wincompatible-pointer-types]
 2536 |     ((ct*) (void *) g_type_check_instance_cast ((GTypeInstance*) ip, gt))
/usr/include/glib-2.0/gobject/gtype.h:526:66: note: in expansion of macro '_G_TYPE_CIC'
  526 | #define G_TYPE_CHECK_INSTANCE_CAST(instance, g_type, c_type)    (_G_TYPE_CIC ((instance), (g_type), c_type))
      |                                                                  ^~~~~~~~~~~
/usr/include/gtk-3.0/gtk/gtkbox.h:40:34: note: in expansion of macro 'G_TYPE_CHECK_INSTANCE_CAST'
   40 | #define GTK_BOX(obj)            (G_TYPE_CHECK_INSTANCE_CAST ((obj), GTK_TYPE_BOX, GtkBox))
      |                                  ^~~~~~~~~~~~~~~~~~~~~~~~~~
gdict-applet.c:368:46: note: in expansion of macro 'GTK_BOX'
  368 |     gtk_button_box_set_child_non_homogeneous(GTK_BOX (bbox), priv->entry, TRUE);
      |                                              ^~~~~~~
In file included from /usr/include/gtk-3.0/gtk/gtk.h:47:
/usr/include/gtk-3.0/gtk/gtkbbox.h:123:75: note: expected 'GtkButtonBox *' {aka 'struct _GtkButtonBox *'} but argument is of type 'GtkBox *' {aka 'struct _GtkBox *'}
  123 | void              gtk_button_box_set_child_non_homogeneous (GtkButtonBox *widget,
      |                                                             ~~~~~~~~~~~~~~^~~~~~
```
In CI the applet is build out-of-process. For that reason i didn't saw that during pull request.

Reply:
Looks like a GtkButtonBox/GtkBox mixup, I'll have to fix that as compilers cannot be relied on to

Reply:
See
https://github.com/mate-desktop/mate-utils/pull/358